modules/firehose/module.go

```diff
@@ -228,33 +237,69 @@ const configSchemaString = `
 `
 
 type Module struct {
-	schema *gjs.Schema
+	schema             *gjs.Schema
+	providerRepository *mongodb.ProviderRepository
 }
 
 func (m *Module) ID() string {
 	return "firehose"
 }
 
-func New() *Module {
+func New(providerRepository *mongodb.ProviderRepository) *Module {
 	schemaLoader := gjs.NewStringLoader(configSchemaString)
 	schema, err := gjs.NewSchema(schemaLoader)
 	if err != nil {
 		return nil
 	}
 	return &Module{
-		schema: schema,
+		schema:             schema,
+		providerRepository: providerRepository,
 	}
 }
 
 func (m *Module) Apply(r *domain.Resource) (domain.ResourceStatus, error) {
+	for _, p := range r.Providers {
+		provider, err := m.providerRepository.GetByURN(p.Urn)
+		if err != nil {
+			return domain.ResourceStatusError, err
+		}
+
+		if provider.Kind == KUBERNETES {
+			releaseConfig := helm.DefaultReleaseConfig()
+			var values = make(map[string]interface{})
+
+			kubeConfig := helm.ToKubeConfig(provider.Configs)
+			helmConfig := &helm.ProviderConfig{
+				Kubernetes: kubeConfig,
+			}
+			helmProvider := helm.NewProvider(helmConfig)
+
+			v := reflect.ValueOf(r.Configs[valuesConfigString])
+			if err := mapstructure.Decode(v, &values); err != nil {
+				return domain.ResourceStatusError, err
+			}
+			releaseConfig.Values = values
+
+			err := mapstructure.Decode(r.Configs, &releaseConfig)
+			if err != nil {
+				return domain.ResourceStatusError, err
+			}
```

Comment:
This can be condensed into one call to `mapstructure.Decode`. Can't we just map `valueConfigString` to `releaseConfig.Values`?

Reply:
Yes!